### ISSUE TITLE
Repository made by factory method @doctrine.orm.default_entity_manager::getRepository($entityName) always have different EntityManager

### ISSUE THREAD

**Issue by fernetmatt** [doctrine/DoctrineBundle#874]
Every repository i configure in my yaml conf file, using the factory method `getRepository()` of `@doctrine.orm.default_entity_manager` get a differente instance of EntityManager causing impossibility to save Entity with mapping associations

I really can't understand why using factory: ["@doctrine.orm.default_entity_manager", getRepository] generate inconsistent EntityManager instances across the repositories while using factory: ["@doctrine.orm.container_repository_factory", getRepository] have better stability. 

TL;DR this story is very long, please go to my SO question here [STACKOVERFLOW](https://stackoverflow.com/q/53093808/928956)

PHP 7.2.7
| Version     | v2.6.1
| Version     | v2.6.2

**Comment by xabbuh** [comment]
I think this is the same as symfony/symfony#28869.

**Comment by alcaeus** [comment]
Looking at the issue @xabbuh linked as well as https://github.com/symfony/symfony/issues/30091 and https://github.com/symfony/symfony/pull/30096, I'd agree that this is not caused by the bundle but rather by the container. Closing here as there's nothing we can fix.
